storm-core/src/jvm/org/apache/storm/validation/ConfigValidation.java

```diff
@@ -493,6 +493,46 @@ public void validateField(String name, Object o) {
         }
     }
 
+    public static class MetricReportersValidator extends Validator {
+
+        @Override
+        public void validateField(String name, Object o) {
+            if(o == null) {
+                return;
+            }
+            SimpleTypeValidator.validateField(name, Map.class, o);
+            if(!((Map) o).containsKey("class") ) {
```

Comment:
Nit: Can we put some of these strings ('class', ''daemons' etc.) in constants somewhere?

Reply:
Done.